File: implementations/nestjs/examples/src/account/application/integration-event/account-suspended-integration-event.ts
// The Integration Event (a public contract) the Account BC exposes to external BCs.
// Kept separate from the internal Domain Event (AccountSuspended) to keep the name/schema
// stable, with an explicit version. The eventName literal is used as the Outbox row's eventType.
export class AccountSuspendedIntegrationEventV1 {
  public readonly eventName = 'account.suspended.v1' as const

  constructor(
    public readonly accountId: string,
    public readonly suspendedAt: string
  ) {}
}

File: implementations/fastapi/examples/src/account/application/integration_event/account_suspended_integration_event.py
from __future__ import annotations

from dataclasses import dataclass
from typing import ClassVar


@dataclass(frozen=True)
class AccountSuspendedIntegrationEventV1:
    """An Integration Event (a public contract) the Account BC exposes to external BCs
    (Card, etc.).

    Kept separate from the internal Domain Event (AccountSuspended) to keep its name/schema
    stable and its version explicit. `event_name` is a ClassVar, so it is not included in
    the `dataclasses.asdict()` payload, and `OutboxWriter.save_all()` uses it as the Outbox
    row's `event_type`.
    """

    event_name: ClassVar[str] = "account.suspended.v1"
    account_id: str
    suspended_at: str
